mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -486,7 +486,8 @@ class LogisticRegression @Since("1.2.0") (
 
   private var optInitialModel: Option[LogisticRegressionModel] = None
 
-  private[spark] def setInitialModel(model: LogisticRegressionModel): this.type = {
+  @Since("3.3.0")
+  def setInitialModel(model: LogisticRegressionModel): this.type = {
```

Comment:
Since `model` is not a `Param`, I think we need to document this difference from other params:
1, `model` is not saved in `MLWritable.save`;
2, `model` is not passed to the trained model `LogisticRegressionModel`

one case is:
```
val lr = new LogisticRegression().setMaxIter(100).setInitialModel(otherModel)
lr.save("/tmp/lr")                                           // does not save initial model
val lr2 = LogisticRegression.load("/tmp/lr")  // does not contains model
```

Reply:
I don't think I'd expect it to be saved - it is just a set of initial conditions

Reply:
ok